cmake/modules/FindMaya.cmake

```diff
@@ -474,6 +482,50 @@ if (MAYA_API_VERSION VERSION_GREATER_EQUAL 20230200)
     message(STATUS "Maya has UFE gizmo drawing")
 endif()
 
+set(MAYA_LINUX_BUILT_WITH_CXX11_ABI FALSE CACHE INTERNAL "MayaLinuxBuiltWithCxx11ABI")
+if(IS_LINUX AND MAYA_Foundation_LIBRARY)
+    # Determine if Maya (on Linux) was built using the new CXX11 ABI.
+    # If yes, then MayaUsd MUST also be built with new ABI.
+    execute_process(
+        COMMAND
+            nm "${MAYA_Foundation_LIBRARY}"
+        COMMAND
+            grep findVariableReplacement
+        COMMAND
+            grep " T "
+       WORKING_DIRECTORY
+            ${MAYA_LIBRARY_DIR}
+        OUTPUT_VARIABLE
+            maya_cxx11_abi)
+    string(FIND ${maya_cxx11_abi} "__cxx1112basic_string" maya_cxx11_abi_index)
+    if(NOT (${maya_cxx11_abi_index} STREQUAL "-1"))
+        set(MAYA_LINUX_BUILT_WITH_CXX11_ABI TRUE CACHE INTERNAL "MayaLinuxBuiltWithCxx11ABI")
+        message(STATUS "Linux: Maya was built with new cxx11 ABI")
+    endif()
+endif()
+
+set(MAYA_MACOSX_BUILT_WITH_UB2 FALSE CACHE INTERNAL "MayaMacOSXBuiltWithUB2")
+if(IS_MACOSX AND MAYA_Foundation_LIBRARY)
+    # Determine if Maya (on OSX) was built with Universal Binary 2 (x86_64 & arm64).
+    # If yes, then MayaUsd can be built with either: Intel, Arm or both.
+    execute_process(
+        COMMAND
+            lipo -archs "${MAYA_Foundation_LIBRARY}"
+        WORKING_DIRECTORY
+            ${MAYA_LIBRARY_DIR}
+        OUTPUT_VARIABLE
+            maya_lipo_output)
+    string(REGEX MATCHALL "(x86_64|arm64)" maya_ub2_match ${maya_lipo_output})
```

Comment:
Use regular expression to see and check list output below to see if both these are present. If they are it means Maya was compiled with UB2 enabled.